### ISSUE TITLE
Check CNR being used in CRP3 reconstruction

### ISSUE THREAD

**Issue by dladams** [DUNE/duneprototypes#14]
We should check the CNR used for CRP3 reco reported here:
https://indico.fnal.gov/event/56948/contributions/253873/attachments/161392/212844/vg_crpcb_confproc.pdf

**Comment by dladams** [comment]
@vgalymov: I see the channel ranges for the VD coldbox have been recorded in the high level config [fcl/vdcoldbox/reco/crpcb_top_process.fcl](https://github.com/DUNE/dunesw/blob/develop/fcl/vdcoldbox/reco/crpcb_top_process.fcl) on lines 11-32. Could we move these to [dunedataprep/DataPrep/fcl/vdcb_tools.fcl](https://github.com/DUNE/dunedataprep/blob/develop/dunedataprep/DataPrep/fcl/vdcb_tools.fcl) without having to add dependency on this high level config an package?

**Comment by vgalymov** [comment]
> I see the channel ranges for the VD coldbox have been recorded in the high level config [fcl/vdcoldbox/reco/crpcb_top_process.fcl](https://github.com/DUNE/dunesw/blob/develop/fcl/vdcoldbox/reco/crpcb_top_process.fcl) on lines 11-32. Could we move these to [dunedataprep/DataPrep/fcl/vdcb_tools.fcl](https://github.com/DUNE/dunedataprep/blob/develop/dunedataprep/DataPrep/fcl/vdcb_tools.fcl) without having to add dependency on this high level config and package?

Yes, totally agree. In retrospect, I should have done this from the beginning, but was in a rush to test the top level fcl.

> I am copying the definitions to vdcb_tools so I can start testing. The names are crpcb_tde_adaX where X={u,v,z}. I think these are valid for CRP3 and not for CRP2. Correct? Should we instead use names crp3cb_tde_adaX?

These are valid for both CRP2 and CRP3. I broke up the the groups by the adapter boards, since there tend to be discontinuities in the coherent noise between the adapter boards. So it is better to apply the filter only to a group of channels within only a given adapter board. The adapter boards are the same for CRP2 and CRP3.

**Comment by dladams** [comment]
Thanks for clarifying. I will add some comments to the fcl. I suppose the old group are only useful for CRP1, it that. Will we have different boards (and thus ranges) for next test and for protoDUNE?

I can make changes in dataprep tomorrow and test those and then we can update dunesw to use those configs. OK?

**Comment by vgalymov** [comment]
> Will we have different boards (and thus ranges) for next test and for protoDUNE?

Normally the adapter boards should be the same. Or more precisely I am not aware that there will be any further design modification in the future. In this case the channel ranges covered by the boards will be the exactly the same except for an offset that will depend on the CRP number in module 0. In the current channel[ mapping scheme](https://github.com/DUNE/dunecore/blob/develop/dunecore/Geometry/CRPChannelMapAlg.cxx), the two bottom CRPs will occupy the 0 - 6143 channel range and the two top will go into 6144 - 12287.  So these channels ranges will have to be offset by 6144 for top CRP A and 9216 for top CRP B.

**Comment by vgalymov** [comment]
> I can make changes in dataprep tomorrow and test those and then we can update dunesw to use those configs. OK?

Sounds good!

**Comment by dladams** [comment]
I am renaming tools vdtcb_XXX to vdtcb1_XXX and vdtcb2_XXX where they differ for data taken with CRP1 and with CRP2 and later.

**Comment by vgalymov** [comment]
> For testing and validation, I want to create an fcl configuration to read CRP3 data and run dataprep only. Should I start from crp3cb_data_oct2022_reco.fcl?

Yes, in principle one should just then change:
```
   produce: @local::crp3_oct22_prod_seq
```
so
```
#include "crp3cb_data_oct2022_reco.fcl" 
...
physics.produce: [caldata]
```
to run only the dataprep

**Comment by dladams** [comment]
I tried that but ran into problems when I also tried changing physics.producers. I ended up using vdcoldbox_raw_dataprep_crp2.fcl. Do you know the difference between these for dataprep only? I think on diff is a bad channel list.

**Comment by vgalymov** [comment]
> Do you know the difference between these for dataprep only? I think on diff is a bad channel list.

Yes, I think it is just the bad channel list. Actually for CRP3 it is empty.

**Comment by dladams** [comment]
I have moved the CNR config from dunesw to dunedataprep. It was bit complicated because I want to retain support for the CRP1 data and many tools were renamed to distinguish between that and CRP2 and beyond. If I push those mods to develop can you check that CRP2/3 reco is working properly?

**Comment by vgalymov** [comment]
yes, sure I can try to do so tomorrow morning.

**Comment by dladams** [comment]
@calcuttj I have (mostly fcl) mods to dunedataprep, duneprototypes and dunesw that need to go in together. Is it ok to push these simultaneously to the develop branches?

**Comment by dladams** [comment]
@vgalymov  I didn't hear back from Jake and so I went ahead and pushed the mods. You should have a CI build in the morning or can build dunedataprep, duneprototypes and dunesw in a local build.

**Comment by vgalymov** [comment]
I fixed up a few little things. This looks to be working now.

**Comment by dladams** [comment]
Glad to her it. Thanks. Changes committed to dunedataprep?

**Comment by vgalymov** [comment]
> Changes committed to dunedataprep?

Yes, I committed all the changes. In dunesw I just needed to change to the new seq name for the dataprep tool list. Since it was rather minor, I pushed the change there directly on develop without the pull request.

**Comment by dladams** [comment]
Looks like the reco code is now in place. To validate this, want to make event displays and for that need a ChannelRange tool for the CPP2+ detectors. That is the topic of [duneprototypes issue 16](https://github.com/DUNE/duneprototypes/issues/16).
